envconfig.go

```diff
@@ -32,75 +54,227 @@ type Decoder interface {
 	Decode(value string) error
 }
 
+// ProcessFunc vistor function definition while traversing the list of options
+type ProcessFunc func(field reflect.Value, tof reflect.StructField, fieldName string, key string, alt_key string, def string, required string) error
+
 func (e *ParseError) Error() string {
 	return fmt.Sprintf("envconfig.Process: assigning %[1]s to %[2]s: converting '%[3]s' to type %[4]s", e.KeyName, e.FieldName, e.Value, e.TypeName)
 }
 
-// Process populates the specified struct based on environment variables
-func Process(prefix string, spec interface{}) error {
+// Visit vists all the configuration options, calling the process function on each
+func Visit(prefix string, spec interface{}, process ProcessFunc) error {
```

Comment:
I kept this public because now people can write "vistor functions" that are external to the core envconfig code, thus if someone comes up with a "cool" new thing, such as display the current configuration, it doesn't have to be merged into envconfig or be done in a private fork.